webapp/app/model/recursive_data.py

```diff
@@ -9,7 +9,22 @@ class PreviousFieldsMissingError(MissingError):
     """Raised in case all the previous fields are missing."""
 
 
-class RecursiveDataModel(BaseModel):
+class PotentialDataModelKeysMixin:
```

Comment:
Maaayybe 
```suggestion
class PotentialDataKeysModelMixin:
```

Reply:
I did it the other way around, because these are the keys of the data model

Reply:
Ahh..sure!